.github/workflows/rust.yml

```diff
@@ -25,6 +25,8 @@ jobs:
       run: sudo apt-get install clang libopenblas-dev libgfortran-9-dev
     - name: Check code formatting
       run: cargo fmt --all -- --check
+    - name: Check cargo clippy warnings
+      run: cargo clippy --all-targets --all-features -- -D warnings
```

Comment:
This compiles the whole project and sadly the results of this compilation are not using during the test. So maybe it worth moving ``fmt`` and ``clippy`` into the separate action workflow. Will probably do in the separate PR